service/src/main/java/com/fecklessweasel/service/objectmodel/UserSession.java

```diff
@@ -72,7 +72,7 @@ public static UserSession resume(Connection connection,
             throw new ServiceException(ServiceStatus.ACCESS_DENIED);
```

Comment:
What do you mean server is down? This is server side code. If user has no auth cookie, this exception is thrown, whether from the REST api or the front-end. If the code is called from the front-end during session resume the exception is caught and the session is set to null.

Effectively, if you attempt to resume session and can't the REST APIs send the access denied message whereas the UI simply has a null session and can decided what to do on a page by page basis.